ArduinoFreevalve/ArduinoFreevalve.ino

```diff
@@ -0,0 +1,74 @@
+// An Arduino program for controlling valve timing on a 6.5ph Predator engine from Harbor Freight.
+//
+// Wesley Kagan, 2020
+// Ric Allinson, 2020
+
+// Pins assignment.
+const int HALL_MAGNET = 2;
+const int EXHAUST_V = 12;
+const int INTAKE_V = 13;
+
+// Control constants.
+const int DEG_PER_MAGNET = 6; // Number of degrees for per magnet.
+
+// Runtime mapping variables.
+bool intakeMap[120] = {};  // Intake open/close mapping from -360 to 360 divided by DEG_PER_MAGNET.
+bool exhaustMap[120] = {}; // Exhaust open/close mapping from -360 to 360 divided by DEG_PER_MAGNET.
+
+// ISR variables.
+volatile int hallCounter;           // The number of magnets after the last TDC.
```

Comment:
Should these be initialized? My 5 minute google research says accessing will give undefined behavior. Not sure what happens when you try to increment or read their values

Reply:
Good catch. The arrays will populated in setup but all the others should be initialized.